tjreports/site/models/reports.php

```diff
@@ -495,7 +508,7 @@ protected function populateState($ordering = '', $direction = 'ASC')
 		$app = JFactory::getApplication();
 		$input = JFactory::getApplication()->input;
 
-		if (!($reportId = $input->get('reportId', 0, 'uint')))
+		if (!($reportId = $input->get('reportId', $input->getString('id'), 'uint')))
```

Comment:
Fetching request data in model looks like a code smell.

Please check this @twsvaishali @manojLondhe

Reply:
API is having id or report as a variable name while the view has report_id. To support all I think we need this code.

Reply:
Well then the id/name should be resolved in the controller or view.html.php and the model should receive only an id right ?

Reply:
ideally yes, but as we have not followed naming conventions earlier and also not while building the APIs this is done

Reply:
Ok get it fixed either way. If a name is passed instead of ID, it should be resolved to the default report id in the controller or in the API plugin itself. Model should always receive id.